File: ASSIGNMENT9/c/Q.1.c
#include <stdio.h>

void findPair(int arr[], int n, int sum) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = i + 1; j < n; j++) {
            if (arr[i] + arr[j] == sum) {
                printf("Pair found (%d, %d)\n", arr[i], arr[j]);
                return;
            }
        }
    }
    printf("No pair found\n");
}

int main() {
    int arr[] = {8, 7, 2, 5, 3, 1};
    int sum = 10;
    int n = sizeof(arr) / sizeof(arr[0]);
    findPair(arr, n, sum);
    return 0;
}

File: ASSIGNMENT9/python/Q.1.py
def find_pair_with_sum(arr, target_sum):
    seen = set()
    for num in arr:
        if target_sum - num in seen:
            return (target_sum - num, num)
        seen.add(num)
    return None

arr = [8, 7, 2, 5, 3, 1]
target_sum = 10
print("Pair with sum {}: {}".format(target_sum, find_pair_with_sum(arr, target_sum)))
